src/App.tsx

```diff
@@ -1,28 +1,24 @@
-import React, { useState, useEffect } from "react"
+import React, { useState } from "react"
 import './App.css';
 import { Routes, Route } from "react-router-dom"
 import Header from "./components/Header"
 import SignUp from "./components/SignUp";
 import SignIn from "./components/SignIn";
 import HomePage from "./components/Home/HomePage"
 
-
 function App() {
-  const [user, setUser] = useState<{ name: string }>({ name: "ini" })
-  // const [user, setUser] = useState(null)
-  if (!user) {
-    return <SignIn />
-  }
+  // const [user, setUser] = useState<{ name: string }>({ name: "ini" })
+  const [user, setUser] = useState(null)
 
   return (
     <div className="App">
       <Header />
       <main>
-        <HomePage />
-
         <Routes>
-          <Route path="/" element={<HomePage />} />
-          <Route path="/signin" element={<SignIn />} />
+          {user ?
+            <Route path="/" element={<HomePage />} /> :
+            <Route path="/signin" element={<SignIn />} />
+          }
           <Route path="/signup" element={<SignUp />} />
```

Comment:
This is incomplete

Reply:
I haven't been able to find what's missing